autorest/utility_test.go

```diff
@@ -234,6 +234,29 @@ func TestStringWithValidString(t *testing.T) {
 	}
 }
 
+func TestStringWithStringSlice(t *testing.T) {
+	s := []string{"string1", "string2"}
+	if String(s, ",") != "string1,string2" {
+		t.Fatal("autorest: String method failed to convert string slice to string")
+	}
+}
+
+type TestEnumType string
+
+func TestStringWithEnum(t *testing.T) {
```

Comment:
[nit] You may consider using [the compound test syntax](https://golang.org/pkg/testing/#T.Run) for these three pretty similar tests.